MKMProvider/tools/makesets.js

```diff
@@ -0,0 +1,9 @@
+const magicSets = require('../data/AllSets.json');
+const fs = require('fs');
+
+var expansions = {};
+Object.keys(magicSets).forEach(el => {
+    expansions[magicSets[el].name] = el;
+});
+
+fs.writeFileSync('data/expansion.json', JSON.stringify(expansions));
```

Comment:
You must specify the absolute path when you're using `fs` methods. Relative path are relative to the user current working dir; it can be differents if you run `node tools/makesets.js` or `cd tools && node makesets.js`.

Use global var `__dirname` to use the directory of the script file.